contracts/Arbitrum_SpokePool.sol

```diff
@@ -3,34 +3,42 @@ pragma solidity ^0.8.0;
 
 import "./SpokePool.sol";
 import "./libraries/CircleCCTPAdapter.sol";
+import "./libraries/OFTTransportAdapter.sol";
 import { CrossDomainAddressUtils } from "./libraries/CrossDomainAddressUtils.sol";
 import { ArbitrumL2ERC20GatewayLike } from "./interfaces/ArbitrumBridge.sol";
 
 /**
  * @notice AVM specific SpokePool. Uses AVM cross-domain-enabled logic to implement admin only access to functions.
  * @custom:security-contact bugs@across.to
  */
-contract Arbitrum_SpokePool is SpokePool, CircleCCTPAdapter {
+contract Arbitrum_SpokePool is SpokePool, CircleCCTPAdapter, OFTTransportAdapter {
     // Address of the Arbitrum L2 token gateway to send funds to L1.
     address public l2GatewayRouter;
 
     // Admin controlled mapping of arbitrum tokens to L1 counterpart. L1 counterpart addresses
     // are necessary params used when bridging tokens to L1.
     mapping(address => address) public whitelistedTokens;
 
+    // A map to store token -> IOFT relationships for OFT bridging.
+    mapping(IERC20 => IOFT) public oftMessengers;
+
     event SetL2GatewayRouter(address indexed newL2GatewayRouter);
     event WhitelistedTokens(address indexed l2Token, address indexed l1Token);
+    event OFTMessengerSet(IERC20 indexed token, IOFT indexed messenger);
```

Comment:
nit: to be consistent, name this `SetOFTMessenger` so the verb comes before the noun